### PR TITLE
[DevDependency] Update ESLint to 8.57.0 & use flat config format

### PR DESCRIPTION
**Pull request by dorian451**
This update includes updating our ESLint config to the new [flat config](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file) format, so that we can benefit from ESLint updates  in the future (like the recently released eslint 9.0). The old config format is deprecated and will probably be removed very soon, so updating our ESLint config to the new format needs to happen at some point soon.

### CODE REVIEW

**Comment by dorian451** [comment]
It appears that a lot of the styling rules we had enabled before this change weren't working properly, 2,443 of the errors in the last js-lint check were style errors caused by updated versions of the rules that we had "enabled" before.

**Comment by dorian451** [comment]
As discussed in our meeting on 6/12/24, I'll defer the updates to the style rules, and put them in another PR.

**Review by williamjallen — COMMENTED** [review]
Can you please update the PR description with information about any behavior changes you've made?  For example, it looks like you've disabled the `no-unused-vars` rule.

**Comment by dorian451** [comment]
That rule was already off before; the new update added it to the list of recommended rules, so there's no behavior change. If I were to enable it,  it would create 293 new linting errors that weren't there with the old config

**Review by williamjallen — CHANGES_REQUESTED** [review]
Good to merge once the typo is addressed.

File: site/eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+const cypress = require('eslint-plugin-cypress/flat');
+
+module.exports = tseslint.config(
+    {
+        name: 'Files to include',
+        files: ['**/*.{js,ts}'],
+    },
+    {
+        // todo: actually fix these files instead of ignoring them
+        ignores: ['cypress/support/e2e.js', 'cypress/plugins/index.js', 'public/js/notebook_builder/widgets/widget.js'],
```
[the remark this answers is not in the document]
Comment (dorian451) [reply site/eslint.config.js]: Fixed in a1e148a903d78a43bf612f0a0644508df1397214

**Review by williamjallen — APPROVED** [review]
Good to merge if tests pass

**Pull request merged**